### DISCUSSION TITLE
ColumnResizerProps TouchEvent/MouseEvent types

### DISCUSSION THREAD

**Discussion by promontis** [TanStack/table#3802] [Ideas]
`ColumnResizerProps` is defined like this..

```
export declare type ColumnResizerProps = {
    title?: string;
    onMouseDown?: (e: MouseEvent) => void;
    onTouchStart?: (e: TouchEvent) => void;
    draggable?: boolean;
    role?: string;
};
```

I think it should be...

```
export declare type ColumnResizerProps = {
    title?: string;
    onMouseDown?: (e: MouseEvent<HTMLElement>) => void;
    onTouchStart?: (e: TouchEvent<HTMLElement>) => void;
    draggable?: boolean;
    role?: string;
};
```

So that the client code can become...

```
 <div {...header.getResizerProps((props: ColumnResizerProps) => ({
```

instead of ...
```
 <div {...header.getResizerProps((props: any) => ({
```

**Comment by promontis** [comment]
If this is ok, I can create a PR

**Comment by tannerlinsley** [comment]
I don’t see the benefit. Can you explain more why they need to change?

Tanner Linsley

**Reply by nrayburn-tech** [reply]
It's likely there are other ways around this or I am missing something, but my use case is being able to spread the return value directly to a component, like `Box` from Material UI.

Modified resizing example to add Material UI box. https://codesandbox.io/s/nifty-meitner-9sg157?file=/src/main.tsx
